### ISSUE TITLE
metaMDS function hangs on metaMDSiter call

### ISSUE THREAD

**Issue by dgiovannelli** [vegandevs/vegan#367]
Dear all,
in the past couple of weeks, I have experienced problems with the metaMDS command. It hangs indefinitely without producing any output. I have tested some old analyses just to repeat the commands that just a few weeks ago were working fine, and the problem is consistent. A similar problem appeared on another two computers we use in the lab, but absent on a third one. The version of vegan and R in all 3 computers is the same (R 3.6.3 and vegan 2.5-6). The three systems experiencing the problem have Ubuntu 20.04 LTS installed, while the one still behaving as expected runs Ubuntu 18.04 LTS. Unfortunately, it is difficult to provide a reproducible example.

I suspect that some of the new libraries are preventing ordinate from running at all, but I cannot figure out how to solve it. 

The two MDS engines called by the function, isoMDS and monoMDS, work as expected on their won. The sqrt and Wisconsin standardization also work on their own.

Using debug() i've seen that the metaMDS function hangs at the metaMDSiter step.

I have also tried to purge R completely and reinstall from scratch, reinstalled vegan and I still have the same problem. Same issue if I install R 4.0. The only thing I have not tried yet is to do a fresh ubuntu 20.04 install. 

Any suggestion?
Thank you,
D

**Comment by dgiovannelli** [comment]
I have tried to use debug on metaMDSiter and found that hangs at this point:

s0 <- switch(engine, monoMDS = monoMDS(dist, y = cmdscale(dist, 
    k = k), k = k, maxit = maxit, ...), isoMDS = isoMDS(dist, 
    k = k, trace = isotrace, maxit = maxit))
Browse[2]> n

**Comment by jarioksa** [comment]
Perhaps a duplicate of issue #366. See first there.

**Comment by jarioksa** [comment]
@dgiovannelli you still have trouble, or can we close this? The problem with major OS or R upgrades is that the **binary** compiled libraries should be concordant with the (1) package, (2) R, and (3) OS libraries. This also means that the **compilers** must be identical in all (C and Fortran – and with **vegan** the latter breaks more often). Please tell us if you still have problems, and with more details, me perhaps can help you.

**Comment by dgiovannelli** [comment]
@jarioksa sorry for the delay. I followed the instruction in #366 and removed completely R and all R related packages. I reinstalled everything but the problem was still present. Not sure how to deal with the different compilers I went nuclear and formatted the entire system (needed anyway). A fresh Ubuntu 20.04 install with R4.0 solved the problem. I'll try a more surgical approach on our other machine, but I might resort to nuking it in case is necessary. Thank you for your help!
